specs/merge/fork-choice.md

```diff
@@ -64,7 +64,8 @@ class PowBlock(Container):
 
 ### `get_pow_block`
 
-Let `get_pow_block(block_hash: Hash32) -> PowBlock` be the function that given the hash of the PoW block returns its data.
+Let `get_pow_block(block_hash: Hash32) -> Optional[PowBlock]` be the function that given the hash of the PoW block returns its data.
+It may result in `None` if the requested block is not found if execution engine is still syncing.
```

Comment:
The consensus spec assumes a 1:1 relationship between CL and EL sync. I'm not sure we should put notes about "syncing" in here. From a consensus spec standpoint, it is unknown if a block is a valid transition block if `get_pow_block` cannot be resolved.

*In practice*, CL will process past the transition point even if EL is syncing, and only later know if such a chain was valid or invalid once EL is done syncing. I'm worried the `assert`s in `on_block` imply too heavily a failure

Reply:
I might be wavering on my resolve here. I don't think we should mention "execution engine" but you could definitely get a block with a terminal block reference that you can't find. Thus there should be an assertion here

Reply:
@djrtwo

is it okay to shorten to:
```suggestion
It may result in `None` if the requested block is not found.
```

Reply:
I edited the above. What do you think?